### PR TITLE
sql generation for updates

### PR DESCRIPTION
**Pull request by spuniani**

### CODE REVIEW

File: src/main/java/com/sarality/db/SQLiteDatabaseProvider.java [hunk 1]
```diff
@@ -63,5 +67,98 @@ public void onCreate(SQLiteDatabase database) {
...
+
+        logger.info("Running upgrade operations for table {} in version {}.", definition.getTableName(), version);
+
+        if (!isSchemaInitialized) {
+          versionColumns = getSchemaForVersion(oldVersion, definition);
+          isSchemaInitialized = true;
+        }
+
+        // is this a new table in this version?
+        List<UpdateOperation> updatesList = operations.get(UpdateOperationType.CREATE_TABLE);
```
Comment (abhideep) [inline src/main/java/com/sarality/db/SQLiteDatabaseProvider.java]: Might make sense to create a SchemaUpdateClassifier to manage the Schema Updates and then use that in both the VersionSchemaGenerator as well as here

**Comment by spuniani** [comment]
Added versionschemaProvider and updated the update method to use the classifier for getting schema updates.

File: src/main/java/com/sarality/db/SQLiteDatabaseProvider.java (hunk 1 above)
Comment (spuniani) [inline src/main/java/com/sarality/db/SQLiteDatabaseProvider.java]: done

File: src/main/java/com/sarality/db/SQLiteDatabaseProvider.java [hunk 2]
```diff
@@ -63,5 +67,98 @@ public void onCreate(SQLiteDatabase database) {
...
+                    updateOperation.getNewColumn()));
+
+          }
+        }
+      }
+    }
   }
+
+
+  private List<Column> getSchemaForVersion(int oldVersion, TableDefinition definition) {
```
Comment (spuniani) [inline src/main/java/com/sarality/db/SQLiteDatabaseProvider.java]: that actually isnt how it was working, but as discussed, makes sense to have a versionSchemaProvider that catalogs the schemas by version and can provide it to the table updater as required. Added the class and modified this method.

**Pull request merged**